src/OutputCollector.jl

```diff
@@ -157,12 +151,12 @@ function wait(collector::OutputCollector)
     # If we've already done this song and dance before, then don't do it again
     if !collector.done
         wait(collector.P)
-        wait(collector.stdout_linestream.task)
-        wait(collector.stderr_linestream.task)
+        fetch(collector.stdout_linestream.task)
+        fetch(collector.stderr_linestream.task)
 
         # Also fetch on any extra tasks we've jimmied onto the end of this guy
         for t in collector.extra_tasks
-            wait(t)
+            fetch(t)
```

Comment:
@Keno @staticfloat, while trying to do `build SpecialFunctions` on 1.0 (osx), I'm seeing BinaryProvider hang on this line when trying to execute the `curl` command to download the binary. I'm not sure what's going on.